features/admin.roles.v2/components/edit-role/edit-role-permission.tsx

```diff
@@ -108,8 +107,11 @@ export const UpdatedRolePermissionDetails: FunctionComponent<RolePermissionDetai
     const [ allAPIResourcesListData, setAllAPIResourcesListData ] = useState<APIResourceInterface[]>([]);
     const [ apiCallNextAfterValue, setAPICallNextAfterValue ] = useState<string>(null);
 
+    const isSharedRole: boolean = role?.properties?.some(
+        (property: RolePropertyInterface) =>
+            property?.name === RoleConstants.IS_SHARED_ROLE && property?.value === "true");
```

Comment:
Let's run this inside useMemo, otherwise this will be executed for every state change in the component.

```suggestion
    const isSharedRole: boolean = useMemo(() => role?.properties?.some(
        (property: RolePropertyInterface) =>
            property?.name === RoleConstants.IS_SHARED_ROLE && property?.value === "true"
     ), [role]);
```

Reply:
Modified